athena-elasticsearch/src/main/java/com/amazonaws/athena/connectors/elasticsearch/ElasticsearchRecordHandler.java

```diff
@@ -133,14 +137,26 @@ protected void readWithConstraint(BlockSpiller spiller, ReadRecordsRequest recor
                                       QueryStatusChecker queryStatusChecker)
             throws RuntimeException
     {
-        logger.info("readWithConstraint - enter - Domain: {}, Index: {}, Mapping: {}",
-                recordsRequest.getTableName().getSchemaName(), recordsRequest.getTableName().getTableName(),
-                recordsRequest.getSchema());
+        String domain;
+        QueryBuilder query;
+        String index;
+        if (recordsRequest.getConstraints().isQueryPassThrough()) {
+            Map<String, String> qptArgs = recordsRequest.getConstraints().getQueryPassthroughArguments();
```

Comment:
please run the verify here as well.

Reply:
done

Reply:
let's discuss offline; just need to understand if query is allowed to be empty here.